open_optical_gating/cli/optical_gater_server.py

```diff
@@ -139,20 +152,11 @@ def analyze(self, pixelArray):
         if self.state == "sync":
             # Using previously-determined reference peiod, analyse brightfield frames
             # to determine predicted trigger time for prospective optical gating
-            (trigger_response, current_phase, current_time_s) = self.sync_state(
-                pixelArray
-            )
-
-            # Logs results and processing time
-            time_fin = time.time()
-            self.timestamp.append(current_time_s)
-            self.phase.append(current_phase)
-            self.processing_rate_fps.append(1 / (time_fin - time_init))
             self.predicted_trigger_time_s.append(
                 None
             )  # placeholder - updated inside sync_state
 
-            return trigger_response, current_phase, current_time_s
```

Comment:
Another important question here: how are you intending the calling code to figure out what the prediction is? Are they just supposed to examine self.predicted_trigger_time_s[-1]? Was there an active reason that you removed the code where this function was returning the predicted time?

Reply:
In fact, I'll strengthen this comment and say I think it **does** need to return the predicted time (or None). Turns out it's not enough to just examine self.predicted_trigger_time_s[-1] - this will initially be an empty list (when we are still establishing the period)

Reply:
Am I correct that you've already solved this with the new 'prediction' variable?

Reply:
TBH, I'm not sure that analyze_pixelarray should be returning anything? The trigger signal that you're using should be dealt with by what is (currently) called `trigger_fluorescence_image_capture`... if you need the prediction time for sending over a socket you should overwrite this method to send it there?